provider/inspec.rb

```diff
@@ -163,32 +139,19 @@ def generate_inspec_test(data, name, target_folder, attribute_file_name)
       )
     end
 
-    def emit_nested_object(data)
-      target = if data[:emit_array]
-                 data[:property].item_type.property_file
+    def emit_nested_object(data, property)
+      target = if property.is_a?(Api::Type::Array)
+                 property.item_type.property_file
                else
-                 data[:property].property_file
+                 property.property_file
                end
       {
         source: File.join('templates', 'inspec', 'nested_object.erb'),
         target: "libraries/#{target}.rb",
-        overrides: emit_nested_object_overrides(data)
+        property: property
       }
     end
 
-    def emit_nested_object_overrides(data)
-      data.clone.merge(
-        api_name: data[:api_name].camelize(:upper),
```

Comment:
We've got this pattern of giving this `data` block a whole bunch of different values. I'm not a huge fan of that pattern. If the value is trivial to figure out, I have no issues letting the template figure it out.